app/js/user.js

```diff
@@ -62,6 +59,24 @@
     });
   }
 
+  function set_credentials(username, password, token) {
+    return asyncStorage.setItem(USER_CREDENTIALS_KEY, {
+      username: username,
+      password: password,
+      token: token
+    });
+  }
+
+  function store_user(username) {
+    return SumoDB.get_user(user.username).then(function(user) {
```

Comment:
If we want to abstract this, it should only do the normalize part and asyncStorage part. We can get a user from `SumoDB.get_user` and `SumoDB.create_user`. That seems like a better level of granularity.